providers/oidc.go

```diff
@@ -87,24 +87,33 @@ func (p *OIDCProvider) EnrichSession(ctx context.Context, s *sessions.SessionSta
 	return nil
 }
 
-// ValidateSession checks that the session's IDToken is still valid
+// ValidateSession checks that the session's id_token or access_token (then a ValidateURL is configured) is still valid
 func (p *OIDCProvider) ValidateSession(ctx context.Context, s *sessions.SessionState) bool {
-	_, err := p.Verifier.Verify(ctx, s.IDToken)
-	if err != nil {
-		logger.Errorf("id_token verification failed: %v", err)
-		return false
+	valid := false
+	// verify session using AccessToken and ValidateURL when set
+	if p.ValidateURL.String() != "" {
+		valid = validateToken(ctx, p, s.AccessToken, makeOIDCHeader(s.AccessToken))
+		if !valid {
+			logger.Errorf("access token is invalid")
+		}
+	} else { // verify ID Token otherwise
+		_, err := p.Verifier.Verify(ctx, s.IDToken)
+		valid = err == nil
+		if !valid {
+			logger.Errorf("id_token verification failed: %v", err)
+		}
 	}
 
-	if p.SkipNonce {
-		return true
+	if !valid || (valid && p.SkipNonce) {
+		return valid
 	}
-	err = p.checkNonce(s)
+	err := p.checkNonce(s)
 	if err != nil {
 		logger.Errorf("nonce verification failed: %v", err)
-		return false
```

Comment:
i personally would prefer the direct return

Reply:
I had a direct return but then code climate complained about too many returns. (see 6cb329f55d5852b27da66fc9283a3b2de0b9df75)  Should I change it back?

Reply:
A yes you are right code climate is quite strict at the moment. Leave it as is for now!